core/mapping/unmarshaler.go

```diff
@@ -384,7 +394,10 @@ func (u *Unmarshaler) processNamedFieldWithValue(field reflect.StructField, valu
 	fieldKind := Deref(field.Type).Kind()
 	switch fieldKind {
 	case reflect.Array, reflect.Map, reflect.Slice, reflect.Struct:
-		return u.processFieldNotFromString(field, value, mapValue, opts, fullName)
+		return u.processFieldNotFromString(field, value, valueWithParent{
+			value:  mapValue,
+			parent: vp.parent,
+		}, opts, fullName)
```

Comment:
Maybe can be used directly the vp variables?

Reply:
fixed.